common-lib/src/main/scala/com/gu/mediaservice/model/ImageMetadata.scala

```diff
@@ -40,13 +40,13 @@ case class ImageMetadata(
     suppliersReference = (that.suppliersReference ++ this.suppliersReference).headOption,
     source = (that.source ++ this.source).headOption,
     specialInstructions = (that.specialInstructions ++ this.specialInstructions).headOption,
-    keywords = that.keywords ++ this.keywords,
+    keywords = (that.keywords ++ this.keywords).headOption,
```

Comment:
I'm not sure `headOption` is right here, its quite different from before, and tbh not sure how it conforms to the type `Option[List[String]]` as i would've thought `headOption` here returns `Option[String]` - I'm sure I'm misunderstanding though

Reply:
The `++` here first does the implicit conversion of `Option[List[String]]` to `Iterable[List[String]]` and _then_ concatenates, so the output will have 0-2 items (`that.keywords ++ this.keywords` ends up being roughly the same as `List(that.keywords, this.keywords).filter(_.isDefined).map(_.get)`), so headOption will give an `Option[List[String]]`

It is quite unlike the old behaviour, but that's because I think the old behaviour was wrong - this func is only used in projection (where `that` is the original metadata and `this` is the user metadata and we're trying to do the merge), and from a quick test: if an image originally had 2 people, one of them was removed by a user, and then image was projected, that other person appears again in the metadata 😬  If subjects or keywords were user editable, I think that projecting could start duplicating items, since they're Lists and not Sets.

This way we should take whatever the user metadata is, and only pick up from the original metadata if no user metadata was set at all. I think this is the correct behaviour, as it makes these fields behave in the same way as all of the other fields and not as exceptions.

Reply:
hmm I wonder if there's a way to make it more explicit, rather than relying on that implicit behaviour (even though it did before)?

Reply:
I've replaced these concats and headOptions with orElse, hopefully that's a bit more straightforward to follow at a glance!